### DISCUSSION TITLE
Flutter support?

### DISCUSSION THREAD

**Discussion by liudonghua123** [Tencent/libpag#45] [Ideas] [answered]
It would be perfect if it supports flutter (android, ios, web, macos, linux, windows) like lottie does. see [lottie](https://pub.dev/packages/lottie), [lottie-flutter](https://github.com/xvrh/lottie-flutter). 

Maybe we can implement it via [`ffi`](https://github.com/dart-lang/ffi) because `libpag` is pure c/c++ native portable code. Or wrap it using classic [`MethodChannel`](https://docs.flutter.dev/development/platform-integration/platform-channels).

**Comment by domchen** [comment, marked as answer]
We haven't implemented the platform interfaces for Flutter yet. But it is actually not difficult to render a PAG file in Flutter. Many apps have done it by themself already through the 'External Texture' of Flutter. What you need is to get an OpenGL texture from Flutter, and then make a PAGSurface from this texture, finally, use a PAGPlayer to render a PAGFile into the PAGSurface. You can give it a try. We also may upload a demo to demonstrate that in recent days.

**Comment by ghost** [comment]
https://github.com/libpag/pag-flutter.git
官方faq找到的，Google半天也没有指向这个仓库，有你的Google。
